File: backend/src/TrailWise.Domain/Entities/PackageTier.cs
using TrailWise.Domain.Enums;

namespace TrailWise.Domain.Entities;

public class PackageTier : BaseEntity
{
    public Guid TourPackageId { get; set; }
    public TourPackage TourPackage { get; set; } = null!;

    public ClassType ClassType { get; set; }
    public bool IncludesFood { get; set; }
    public decimal BasePricePerPerson { get; set; }
    public bool RequiresAC { get; set; }

    public ICollection<Booking> Bookings { get; set; } = new List<Booking>();
}

File: frontend-mobile/lib/models/package_tier.dart
class PackageTier {
  final String id;
  final String classType;
  final bool includesFood;
  final double basePricePerPerson;
  final bool requiresAC;

  PackageTier({
    required this.id,
    required this.classType,
    required this.includesFood,
    required this.basePricePerPerson,
    required this.requiresAC,
  });

  factory PackageTier.fromJson(Map<String, dynamic> json) => PackageTier(
        id: json['id'] as String,
        classType: json['classType'] as String,
        includesFood: json['includesFood'] as bool,
        basePricePerPerson: (json['basePricePerPerson'] as num).toDouble(),
        requiresAC: json['requiresAC'] as bool,
      );
}
